state/entrySyncing.go

```diff
@@ -204,11 +205,13 @@ func (s *State) GoSyncEntries() {
 		firstMissing = -1
 
 	dirblkSearch:
-		for scan := start; scan <= s.GetHighestSavedBlk(); scan++ {
+		for scan := start; scan <= highestSavedBlk; scan++ {
 
 			if firstMissing < 0 {
 				if scan > 1 {
-					s.EntryDBHeightComplete = scan - 1
+					if s.EntryDBHeightComplete < scan-1 {
```

Comment:
Check if  scan -1 is actually greater than the current value of EntryDBHeightComplete to ensure it doesn't time-travel.